float32 th1
float32 th2
float32 th3
float32 th4
float32 angle1
float32 angle2
float32 angle3
float32 angle4
